src/lib/OpenEXR/ImfCompositeDeepScanLine.cpp

```diff
@@ -508,6 +525,15 @@ CompositeDeepScanLine::readPixels (int start, int end)
         overall_sample_count += total_sizes[ptr];
     }
 
+    if (maximumSampleCount >=0 &&  overall_sample_count > maximumSampleCount)
```

Comment:
is 0 a valid max? if you pay attention to my above suggestion, this should probably be > 0

Reply:
My reasoning was that setting the limit to 0 would be a way of completely disabling the transparent reading of deep images, and we might want to add better checks for this later on, so exceptions are thrown from the constructor rather than from readPixels if maximumSampleCount==0. The alternative approach might be to encourage code to check the `type` header attribute before reading pixels.